curator-framework/src/main/java/org/apache/curator/framework/imps/GzipCompressionProvider.java

```diff
@@ -45,6 +45,23 @@
     /** GZIP header magic number. */
     private static final int GZIP_MAGIC = 0x8b1f;
 
+    private static final String JAVA_VERSION_STR = System.getProperty("java.version", "-1");
+    private static final int JAVA_VERSION;
+    static {
+        if ("1.8".equals(JAVA_VERSION_STR)) {
+            JAVA_VERSION = 8;
+        } else if (JAVA_VERSION_STR.contains("-")) {
+            // e.g. 18-ea, take 18
+            JAVA_VERSION = Integer.parseInt(JAVA_VERSION_STR.split("-", 2)[0]);
+        } else if (JAVA_VERSION_STR.contains(".")) {
+            // e.g. 17.0.1, take 17
+            JAVA_VERSION = Integer.parseInt(JAVA_VERSION_STR.split("\\.", 2)[0]);
+        } else {
+            JAVA_VERSION = Integer.parseInt(JAVA_VERSION_STR, 10);
+        }
+    }
+    private static final byte OS_BIT = JAVA_VERSION >= 16 ? (byte) -1 : 0;
```

Comment:
could you add a comment explaining why this is needed ?

Reply:
I just reverse engineered it.
I don't know what exactly has changed in JDK 16 that caused the breakage here.
This is the reason I said `I'd recommend a deeper look here!`. Someone who knows this code better may have better idea what is going on.

Reply:
I think this is the related issue 
https://bugs.openjdk.java.net/browse/JDK-8244706

Reply:
I will add it as a comment!